homeassistant/components/hifiberry/config_flow.py

```diff
@@ -0,0 +1,98 @@
+"""Config flow for hifiberry integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from pyhifiberry.audiocontrol2 import Audiocontrol2, Audiocontrol2Exception
+import voluptuous as vol
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.typing import DiscoveryInfoType
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]):
+    """Validate the user input allows us to connect."""
+
+    api = Audiocontrol2(
+        async_get_clientsession(hass), host=data["host"], port=data["port"]
+    )
+
+    try:
+        await api.metadata()
+    except Audiocontrol2Exception as err:
+        raise CannotConnect from err
+    except Exception as err:
+        raise Exception from err
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for hifiberry."""
+
+    VERSION = 1
+
+    def __init__(self) -> None:
+        """Initialize with default hostname."""
+        self.host = "hifiberry.local"
+
+    @property
+    def schema(self):
+        """Return current schema."""
+        return vol.Schema(
+            {
+                vol.Required("host", default=self.host): str,
```

Comment:
If a user changes the value of host, then clicks submit, then there is an error and the form is re-renderd it seems like the value they entered for the host will be overridden with this host. Am I reading this right?